org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/client/IRepl.java

```diff
@@ -22,17 +23,20 @@
      *
      * @param input
      *            The input to send for evaluation.
+     * @return An {@link IHook} to process the result of the evaluation, or {@code null} when an
+     *         empty input would be executed.
      * @throws MetaborgException
      *             When something goes wrong during execution.
      * @throws CommandNotFoundException
      *             When the command could not be found.
      */
-    default void eval(String input) throws MetaborgException, CommandNotFoundException {
+    // TODO: return Optional?
+    default IHook eval(String input) throws MetaborgException, CommandNotFoundException {
         input = input.trim();
         if (input.length() == 0) {
-            return;
+            return null;
```

Comment:
I still have to update the clients to handle this. Perhaps it'd be better to return an `Optional` instead.